internal/server/storage/dbstorage.go

```diff
@@ -0,0 +1,170 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/jackc/pgerrcode"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+
+	"github.com/npavlov/go-metrics-service/internal/domain"
+	"github.com/npavlov/go-metrics-service/internal/model"
+)
+
+type DBStorage struct {
+	db *gorm.DB
+}
+
+func NewDBStorage(db *gorm.DB) *DBStorage {
+	return &DBStorage{
+		db: db,
+	}
+}
+
+func retryOperation(operation func() error) error {
+	maxRetries := 3
+	retryDelays := []time.Duration{1 * time.Second, 3 * time.Second, 5 * time.Second}
```

Comment:
[CHANGE REQUEST] If I were you, I would use `Backoff`
https://pkg.go.dev/github.com/cenkalti/backoff/v4#section-readme

Reply:
Added to implementation